src/prefect/utilities/storage.py

```diff
@@ -102,15 +105,37 @@ def extract_flow_from_module(module_str: str, flow_name: str = None) -> "Flow":
     """
 
     # load the module
-    module = importlib.import_module(module_str)
+    module_parts = module_str.split(":", 2)
+
+    if len(module_parts) == 2 and flow_name is not None:
+        raise ValueError(
+            "Provide either `module_str` without an attribute specifier or remove `flow_name`."
+        )
+
+    module_name, flow_name = (
+        module_parts if len(module_parts) == 2 else [module_str, flow_name or ""]
+    )
```

Comment:
Would you mind de-clevering this logic here, I find the one-liner a bit hard to follow. Merging this with the `if len(module_parts) == 2` logic above would make things clearer IMO.